api/src/modules/h3-data/h3-data.service.ts

```diff
@@ -163,20 +168,49 @@ export class H3DataService {
     /**
      * @note To generate a Risk Map, harvest data and h3Data by indicatorId are required
      */
+
+    const availableIndicatorYears: number[] =
+      await this.getAvailableYearsForH3IndicatorData(indicatorId);
+
+    const indicatorDataYear: number | undefined =
+      availableIndicatorYears.includes(year)
+        ? year
+        : availableIndicatorYears.find((el) => el < year);
```

Comment:
What would happen here if we actually have data for the requested year? Wouldn't we performing calcs with for the previous year?

`const arr = [2004, 2003, 2002, 2001, 2000]

 const year = 2002

 const availableYear = arr.find((el) => el < year)

 console.log(availableYear) // 2001`